File: gertonargent_app/lib/data/models/planned_purchase.dart
class PlannedPurchase {
  final int id;
  final int userId;
  final String name;
  final String? description;
  final double amount;
  final String category;
  final DateTime? plannedDate;
  final DateTime createdAt;
  final DateTime? purchasedAt;
  final String status; // 'pending', 'purchased', 'cancelled'
  final int? transactionId;

  PlannedPurchase({
    required this.id,
    required this.userId,
    required this.name,
    this.description,
    required this.amount,
    required this.category,
    this.plannedDate,
    required this.createdAt,
    this.purchasedAt,
    required this.status,
    this.transactionId,
  });

  factory PlannedPurchase.fromJson(Map<String, dynamic> json) {
    return PlannedPurchase(
      id: json['id'],
      userId: json['user_id'],
      name: json['name'],
      description: json['description'],
      amount: (json['amount'] as num).toDouble(),
      category: json['category'],
      plannedDate: json['planned_date'] != null
          ? DateTime.parse(json['planned_date'])
          : null,
      createdAt: DateTime.parse(json['created_at']),
      purchasedAt: json['purchased_at'] != null
          ? DateTime.parse(json['purchased_at'])
          : null,
      status: json['status'],
      transactionId: json['transaction_id'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'description': description,
      'amount': amount,
      'category': category,
      'planned_date': plannedDate?.toIso8601String(),
    };
  }

  bool get isPending => status == 'pending';
  bool get isPurchased => status == 'purchased';
  bool get isCancelled => status == 'cancelled';

  String get categoryEmoji {
    switch (category.toLowerCase()) {
      case 'alimentation':
        return '🍔';
      case 'transport':
        return '🚗';
      case 'logement':
        return '🏠';
      case 'sante':
        return '🏥';
      case 'education':
        return '📚';
      case 'loisirs':
        return '🎮';
      case 'vetements':
        return '👕';
      case 'communication':
        return '📱';
      case 'epargne':
        return '💰';
      default:
        return '📦';
    }
  }

  String get statusLabel {
    switch (status) {
      case 'pending':
        return 'En attente';
      case 'purchased':
        return 'Acheté';
      case 'cancelled':
        return 'Annulé';
      default:
        return status;
    }
  }
}

File: backend/app/models/planned_purchase.py
from sqlalchemy import Column, Integer, String, Float, DateTime, Boolean, ForeignKey, Enum as SQLEnum
from sqlalchemy.orm import relationship
from datetime import datetime
import enum
from .database import Base

class PurchaseStatus(str, enum.Enum):
    """
    Cycle de vie d'un achat planifié.
    Permet de distinguer les envies en attente des dépenses réalisées.
    """
    PENDING = "pending"      # En attente de validation ou d'argent
    PURCHASED = "purchased"  # Converti en transaction réelle
    CANCELLED = "cancelled"  # Abandonné pour économiser

class CategoryEnum(str, enum.Enum):
    """Catégories de dépenses"""
    ALIMENTATION = "alimentation"
    TRANSPORT = "transport"
    LOGEMENT = "logement"
    SANTE = "sante"
    EDUCATION = "education"
    LOISIRS = "loisirs"
    EPARGNE = "epargne"
    VETEMENTS = "vetements"
    COMMUNICATION = "communication"
    AUTRE = "autre"

class PlannedPurchase(Base):
    """
    Liste de souhaits ("Wishlist") financière.
    Aide l'utilisateur à différer ses achats pour éviter l'impulsivité.
    """
    __tablename__ = "planned_purchases"

    id = Column(Integer, primary_key=True, index=True)
    user_id = Column(Integer, ForeignKey("users.id"), nullable=False)
    
    # Informations sur l'achat
    name = Column(String, nullable=False)  # Ex: "Acheter des chaussures"
    description = Column(String, nullable=True)
    amount = Column(Float, nullable=False)  # Montant prévu
    category = Column(SQLEnum(CategoryEnum), nullable=False)
    
    # Dates
    planned_date = Column(DateTime, nullable=True)  # Date prévue pour l'achat
    created_at = Column(DateTime, default=datetime.utcnow)
    purchased_at = Column(DateTime, nullable=True)  # Date réelle d'achat
    
    # Statut
    status = Column(SQLEnum(PurchaseStatus), default=PurchaseStatus.PENDING)
    
    # Lien avec la transaction réelle (si déjà acheté)
    transaction_id = Column(Integer, ForeignKey("transactions.id"), nullable=True)
    
    # Relations
    user = relationship("User", back_populates="planned_purchases")
    transaction = relationship("Transaction", back_populates="planned_purchase")

    def __repr__(self):
        return f"<PlannedPurchase {self.name} - {self.amount} FCFA ({self.status})>"
